src/databricks/labs/remorph/helpers/dashboard_publisher.py

```diff
@@ -0,0 +1,41 @@
+import json
+from pathlib import Path
+from typing import Any
+
+from databricks.labs.blueprint.installation import Installation
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ObjectInfo
+
+
+class DashboardPublisher:
+    def __init__(self, ws: WorkspaceClient, installation: Installation):
+        self._ws = ws
+        self._installation = installation
+
+    def create(self, dashboard_file: Path, parameters: dict[str, Any] | None = None) -> ObjectInfo:
+        if parameters is None:
+            parameters = {}
+
+        dashboard_data = self._substitute_params(dashboard_file, parameters)
+        self._installation.upload(dashboard_file.name, dashboard_data)
```

Comment:
I think we should add `try` and `except` blocks since we're interacting with a remote DB workspace.

Reply:
@bishwajit-db can you fix this.

Reply:
IMO, we can add retry logic based on databricks.sdk.retries.retried, but we should keep the exception handling to the calling code. Handling the exception here will either swallow it or duplicate handling in both here and the caller. I plan to add that to the installer. What do you say, @sundarshankar89 @vijaypavann-db?